docs/start/getting-started/fragments/angular/data-model.md

```diff
@@ -119,69 +119,114 @@ Update `src/tsconfig.app.json` to include the "node" compiler option in *types*:
 In your `src/app/app.component.ts` file, add the following imports and modifications to your class to to add data to your database with a mutation by using the `API.service` file which was generated when you ran `amplify add api`:
 
 ```javascript
-import { APIService } from './API.service';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { APIService } from '../API.service';
 
-export class AppComponent {
+@Component(...)
+export class AppComponent implements OnInit {
+  public createForm: FormGroup;
 
-  constructor(private apiService: APIService) {}
+  constructor(private api: APIService, private fb: FormBuilder) { }
 
-  createTodo() {
-    this.apiService.CreateTodo({
-        name: 'Angular',
-        description: 'testing'
+  async ngOnInit() {
+    this.createForm = this.fb.group({
+      'name': ['', Validators.required],
+      'description': ['', Validators.required],
+      'city': ['', Validators.required]
+    });
+  } 
+  
+  public onCreate(restaurant: any) {
+    this.api.CreateRestaurant(restaurant).then(event => {
+      console.log('item created!');
+      this.createForm.reset();
+    })
+    .catch(e => {
+      console.log('error creating restaurant...', e);
     });
   }
 }
 ```
 
-Add the following to your `src/app/app.component.html` to add a button which calls your `createTodo()` method:
+Add the following to your `src/app/app.component.html` to add the form to create a new restaurant:
 
 ```html
-<button (click)="createTodo()">Add Todo</button>
+<div class="form-body">
+  <form autocomplete="off" [formGroup]="createForm" (ngSubmit)="onCreate(createForm.value)">
+    <div>
+      <label>Name: </label>
+      <input type="text" formControlName="name" autocomplete="off">
+    </div>
+    <div>
+      <label>Description: </label>
+      <input type="text" formControlName="description" autocomplete="off">
+    </div>
+    <div>
+      <label>City: </label>
+      <input type="text" formControlName="city" autocomplete="off">
+    </div>
+    <button type="submit">Submit</button>
+  </form>
+</div>
 ```
 
-Next, update the class to list all items in the database by running a `ListTodos` query when the app starts by implementing [OnInit](https://angular.io/api/core/OnInit) and storing the items in an array:
+Create a file at `types/restaurant.ts` to define a type for the Restaurant:
+
+```ts
+export type Restaurant = {
+  id : string,
+  name : string,
+  description : string,
+  city: string
+}; 
+```
+
+Next, update the class to list all items in the database by running a `ListRestaurants` query when the app starts by implementing [OnInit](https://angular.io/api/core/OnInit) and storing the items in an array:
```

Comment:
```suggestion
Next, update your `AppComponent` class so that it will list all restaurants in the database when the app starts. To do so, implement [OnInit](https://angular.io/api/core/OnInit) add a `ListRestaurants` query. Store the query results in an array.
```